harvester/pmh_interface.py

```diff
@@ -0,0 +1,28 @@
+"""
+Fetch data from an OAI-PMH API of Metashare
+"""
+
+from sickle import Sickle
+
+
+class PMH_API:
+    """
+    Interface for fetching data from an OAI-PMH API
+    """
+
+    def __init__(self, url):
+        """
+        :param url: URL of the OAI-PMH API used
+        """
+        self.sickle = Sickle(url)
+
+    def get_all_metadata_records(self):
+        """
+        Get content for all records available from the API.
+        """
+        records = self.sickle.ListRecords(metadataPrefix="info")
+        metadata_records_list = []
+        for record in records:
+            metadata_records_list.append(record)
+        return metadata_records_list
```

Comment:
Is it necessary/a good idea to convert the generator that sickle returns to a list? You probably know best what kind of purposes this function will be used for.

If a list is what you want to return, you can convert the result of `ListRecords` to a list like this:

```
records = list(self.sickle.ListRecords(metadataPrefix="info"))
```
or alternatively only wrap into a list when returning:
```
return list(records)
```

So there's no need for a loop.

Reply:
I just realized that the data type that sickle returns is not a normal generator after all, so my previous comment is pretty much useless :D unless you want to actually return a generator, in which case you could use `yield` in the loop. But if list is what is wanted, this is fine.